pkg/asset/agent/installconfig.go

```diff
@@ -159,6 +159,9 @@ func (a *OptionalInstallConfig) validateSNOConfiguration(installConfig *types.In
 		if *installConfig.ControlPlane.Replicas != 1 {
 			fieldPath = field.NewPath("ControlPlane", "Replicas")
 			allErrs = append(allErrs, field.Required(fieldPath, fmt.Sprintf("control plane replicas must be 1 for %s platform. Found %v", none.Name, *installConfig.ControlPlane.Replicas)))
+		} else if len(installConfig.Compute) == 0 {
```

Comment:
Shouldn't this be `!= 0`?

Reply:
Sorry, I missed the `len()`.

Given that the user's intention is clear, shouldn't we just default this to a config section with no replicas rather than throw an error at them?

Where is it currently being defaulted to a higher number of compute replicas if no explicit config is provided?

Reply:
IINW the default for the worker is set [here](https://github.com/openshift/installer/blob/e1959c2931b2cee35c789958cc60989e10ee285d/pkg/types/defaults/installconfig.go#L74-L79).

Personally I'd like the defaulting if the install-config would have a field to specify explicitly the desired configuration (ie, `platformConfig: SingleNode`) - without it, the error message seems a less risky option.